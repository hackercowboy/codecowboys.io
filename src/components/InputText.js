import React from 'react';
import PropTypes from 'prop-types';

import { Field } from 'formik';
import ErrorMessage from './ErrorMessage';

function InputText({
  name, placeholder, disabled,
}) {
  return (
    <div className="input-text">
      <Field
        type="text"
        name={name}
        disabled={disabled}
        placeholder={placeholder}
      />
      <ErrorMessage name={name} />
    </div>
  );
}

InputText.propTypes = {
  name: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  placeholder: PropTypes.string.isRequired,
};

InputText.defaultProps = {
  disabled: false,
};

export default InputText;
