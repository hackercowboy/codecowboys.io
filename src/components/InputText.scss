.input-text {
  height: 70px;
  
  input {
    height: 40px;
    padding: 10px;
  }
}
