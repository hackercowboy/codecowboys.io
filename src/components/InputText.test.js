import React from 'react';
import { shallow } from 'enzyme';

import InputText from './InputText';

describe('<InputText/>', () => {
  it('should render', () => {
    const wrapper = shallow(<InputText name="test" placeholder="Test" />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
