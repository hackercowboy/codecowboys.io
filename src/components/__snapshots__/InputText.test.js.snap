// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText/> should render 1`] = `
"<div className=\\"input-text\\">
  <Field type=\\"text\\" name=\\"test\\" placeholder=\\"Test\\" />
  <FormikConnect(ErrorMessageImpl) name=\\"test\\" />
</div>"
`;
